### PR TITLE
GET /api/v1/patients

### PR DESCRIPTION
**Pull request by Alaalser**
Relates #11

## GET /api/v1/patients
- Create query 
- Create controllers
- Add its route
- Create the route test

### CODE REVIEW

File: server/test/index.test.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
+const request = require('supertest');
+const app = require('../app');
+const dbBuild = require('../database/config/build');
+const connection = require('../database/config/connection');
+
+beforeEach(() => dbBuild());
```
Comment (hassanelnajjar) [inline server/test/index.test.js:6]: Hi Alaa,Please put beforeEach and After Each inside the describe

File: server/test/index.test.js [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+afterAll(() => connection.end());
+
+describe('Get all patients', () => {
+  test('Route /patients status 200, json header', async () => {
```
Comment (hassanelnajjar) [inline server/test/index.test.js:10]: Also,please make a new test for the database query

File: server/test/index.test.js [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+    expect.assertions(1);
+    const res = await request(app)
+      .get('/api/v1/patients')
+      .expect(200)
+      .expect('Content-Type', /json/);
+    expect(res.body.data).toHaveLength(20);
```
Comment (hassanelnajjar) [inline server/test/index.test.js:16]: you can add return here before expect like this and remove the expect.assertions(1);
```js
return expect(res.body.data).toHaveLength(20);
```

File: package.json [hunk 4]
```diff
@@ -57,4 +57,4 @@
     "eslint-plugin-prettier": "^3.3.1",
     "prettier": "2.2.1"
   }
-}
+}
```
Comment (hassanelnajjar) [inline package.json:60]: Please Add Line at the end.

File: .circleci/config.yml [hunk 5]
```diff
@@ -1,7 +1,70 @@
...
+      - run: npm test
 workflows:
-  node-tests:
-    jobs:
-      - node/test
+  version: 2
+  build_and_test: # this is the name of our workflow
+    jobs: # and here we list the jobs we are going to run.
+      - build
+      - test
```
Comment (hassanelnajjar) [inline .circleci/config.yml:70]: Please Add Line at the end.

**Review by hassanelnajjar — APPROVED** [review]

**Review by hassanelnajjar — APPROVED** [review]

File: .circleci/config.yml [hunk 6]
```diff
@@ -67,4 +67,5 @@ workflows:
   build_and_test: # this is the name of our workflow
     jobs: # and here we list the jobs we are going to run.
       - build
-      - test
\ No newline at end of file
+      - test
+
```
Comment (Mu7ammadAbed) [inline .circleci/config.yml:71]: Fix this

**Review by hassanelnajjar — APPROVED** [review]

File: server/routes/patients/index.js [hunk 7]
```diff
@@ -1,9 +1,16 @@
...
+  getAllProfileData,
+  getPatientByNameOrPhone,
+} = require('../../controllers');
 
 router.get('/', getPatients);
 router.get('/search');
 router.route('/:patientId').get().patch();
+router.get('/');
+router.get('/search', getPatientByNameOrPhone);
+router.route('/:patientId').get(getAllProfileData).patch();
```
Comment (Mu7ammadAbed) [inline server/routes/patients/index.js]: Remove the repeated lines

**Review by Mu7ammadAbed — CHANGES_REQUESTED** [review]

**Review by Mu7ammadAbed — APPROVED** [review]

**Pull request merged**